File: java/advent2022/src/main/java/org/mersic/Day03.java
package org.mersic;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public class Day03 {
    
    private static int score(char c) {
        if (c <= 'Z') {
            return (c-'A') + 27;
        } else {
            return (c-'a') + 1;
        }
    }
    
    public static void main(String args[]) throws Exception {
        List<String> input = Files.readAllLines(Path.of(Day03.class.getClassLoader().getResource("day.03.input").toURI()));
        
        int totalPriority = 0;
        for (String s : input) {
            String s1 = s.substring(0, s.length()/2);
            String s2 = s.substring(s.length()/2);
            for (char c : s1.toCharArray()) {
                if (s2.indexOf(c) >= 0) {
                    totalPriority += score(c);
                    break;
                }
            }
        }
        
        int totalPriority2 = 0;
        for (int i = 0; i < input.size(); i+=3) {
            String s1 = input.get(i);
            String s2 = input.get(i+1);
            String s3 = input.get(i+2);

            for (char c : s1.toCharArray()) {
                if (s2.indexOf(c) >= 0 && s3.indexOf(c) >= 0) {
                    totalPriority2 += score(c);
                    break;
                }
            }
        }
        
        System.out.println("day 3 part 1: " + totalPriority);
        System.out.println("day 3 part 2: " + totalPriority2);
    }
    
    
}

File: kotlin/advent2022/src/main/kotlin/Day03.kt
import java.nio.file.Files
import java.nio.file.Path

fun score(c : Char): Int {
    if (c <= 'Z') {
        return c - 'A' + 27
    } else {
        return c - 'a' + 1
    }
}

fun main() {
    val lines = Files.readAllLines(Path.of(Stub().javaClass.getClassLoader().getResource("day.03.input").toURI()))
    
    var totalPriority = 0
    for (line in lines) {
        val l1 = line.substring(0, line.length/2)
        val l2 = line.substring(line.length/2)
        
        for (c in l1) {
            if (l2.contains(c)) {
                totalPriority += score(c)
                break
            }
        }
    }
    
    var totalPriority2 = 0
    for (i in lines.indices step 3) {
        val l1 = lines[i]
        val l2 = lines[i+1]
        val l3 = lines[i+2]
        
        for (c in l1) {
            if (l2.contains(c) && l3.contains(c)) {
                totalPriority2 += score(c)
                break
            }
        }
    }
    
    println("day 3 part 1: $totalPriority")
    println("day 3 part 2: $totalPriority2")
}